src/mercury-agent/mercury/agent/procedures/hp_firmware.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2015 Jared Rodriguez (jared.rodriguez@rackspace.com)
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import glob
+import logging
+import lxml
+import os
+import requests
+
+from mercury.agent.capabilities import capability
+from mercury.agent.procedures.firmware import HPFirmwareException
+from mercury.agent.hardware.platform.platform_detection import is_hp
+from mercury.common.helpers import cli
+
+log = logging.getLogger(__name__)
+firmware_path = '/tmp/hp/firmware'
+
+
+def _extract(tarball_path, extract_path):
+    os.makedirs(extract_path)
+    log.info('Extracting {0}'.format(tarball_path))
+    cmd = 'tar --strip-components=1 -xvf {0} -C {1}'.format(tarball_path,
+                                                            extract_path)
+
+    result = cli.run(cmd)
+    if result.returncode:
+        log.error('Could not extract archive: {0}'.format(result))
+        return False
+
+    return True
+
+
+def parse_fw_report(xml_file):
+    fw_tree = lxml.etree.parse(xml_file)
+    component_list = []
+    # For checking needs_install XML tag is present or not
+    component_list += parse_xml_for_components(fw_tree, 'needs_install')
```

Comment:
`component_list = parse_xml_for_components(fw_tree, 'needs_install')`

Don't need first ` component_list = []` as it's already returned as a list